xl2times/transforms.py

```diff
@@ -570,6 +570,69 @@ def process_user_constraint_table(
     return [process_user_constraint_table(t) for t in tables]
 
 
+def generate_uc_properties(
+    config: datatypes.Config,
+    tables: List[datatypes.EmbeddedXlTable],
+    model: datatypes.TimesModel,
+) -> List[datatypes.EmbeddedXlTable]:
+    """
+    Generate a dataframe containing User Constraint properties
+    """
+
+    uc_tables = [table for table in tables if table.tag == datatypes.Tag.uc_t]
+    columns = ["uc_n", "uc_desc", "region", "reg_action", "ts_action"]
+    user_constraints = pd.DataFrame(columns=columns)
+
+    for uc_table in uc_tables:
+        df = uc_table.dataframe.loc[:, ["uc_n"]].drop_duplicates(keep="first")
+        df = df.merge(
+            uc_table.dataframe.loc[:, ["uc_n", "uc_desc"]]
+            .drop_duplicates(keep="first")
+            .dropna(),
+            how="left",
+        )
+
+        for key in uc_table.uc_sets.keys():
+            if key.startswith("R_"):
+                df["reg_action"] = key
+                df["region"] = uc_table.uc_sets[key].upper().strip()
+            elif key.startswith("T_"):
+                df["ts_action"] = key
```

Comment:
It is because `uc_sets` is a dictionary and there should be only single keys that fit `R_*` and "T_*" patterns.
I can look into ensuring this in a separate PR.